### PR TITLE
SAK-32508 Cache failed lookups for a shorter time.

### PR DESCRIPTION
**Pull request by buckett**
Also added support for recording an error about why the failure to load an entry occurred, although this isn’t yet shown in the UI.
Removed the ExternalCalendarSubscriptionService cover and updated to use the service, there wasn’t any use of the cover outside the calendar tool.
Split out the preloading of calendars into it’s own bean as this makes testing of the calendar service easier.
Add generics and update to Java 8 for parts of the code I was looking at.
Also drop needless methods from API.

### CODE REVIEW

File: calendar/calendar-api/api/src/java/org/sakaiproject/calendar/api/ExternalCalendarSubscriptionService.java [hunk 1]
```diff
@@ -90,12 +90,12 @@
...
 	 */
-	public Set<ExternalSubscription> getAvailableInstitutionalSubscriptionsForChannel(
+	public Set<ExternalSubscriptionDetails> getAvailableInstitutionalSubscriptionsForChannel(
 			String reference);
 
 	/** Get list of calendar subscriptions for a given Calendar channel */
-	public Set<ExternalSubscription> getSubscriptionsForChannel(String reference,
-			boolean loadCalendar);
+	public Set<ExternalSubscriptionDetails> getSubscriptionsForChannel(String reference,
+																	   boolean loadCalendar);
```
Comment (master-bob) [inline calendar/calendar-api/api/src/java/org/sakaiproject/calendar/api/ExternalCalendarSubscriptionService.java]: These extra tabs seem unneeded.

**Comment by buckett** [comment]
I need to fixup the tests before merging.

**Comment by buckett** [comment]
The attachment of the JAR artifact should be part of the sakai maven plugin really.

**Comment by ottenhoff** [comment]
I'm not following the TravisCI failure here.....

**Comment by buckett** [comment]
I didn't do a `mvn clean` locally so the path to the extra artifact to archive was wrong and good old travis caught it. Fixed and it should work this time. Builds locally with:

    mvn clean install -pl calendar/calendar-impl/impl -am

**Pull request merged**